public/scripts/main.js

```diff
@@ -51,18 +58,27 @@ function* interpolateColors(dataLength, scale, range) {
 function createChart(elementId, type = "bar", payload = {}) {
   const { labels, data, interpolate = d3.interpolateCool } = payload;
   const options = JSON.parse(JSON.stringify(kChartOptions));
+  if (type === "horizontalBar") {
+    type = "bar";
+    options.indexAxis = "y";
+  }
   if (type === "pie") {
     options.legend.display = true;
     options.plugins.datalabels.align = "center";
-  }
-  else {
+  } else {
+    options.plugins.legend.display = false;
```

Comment:
```suggestion
  } 
  else {
    options.plugins.legend.display = false;
```

You need to check the linter is OK ;)
There are other small problems, you can run `npx eslint src public` if your IDE is not configured (if you install ESLint extension you should not miss theses errors, i guess you're using VSC)

Reply:
I I corrected that